mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java

```diff
@@ -122,6 +123,79 @@ public StatisticResult addAssists(Transaction transaction, int amount) {
         return new StatisticResult(newValue);
     }
 
+    /**
+     * Adds experience to the user account. Infallible.
+     * @param transaction represents the transaction
+     * @param amount represents the amount of xp to add
+     * @return a result with the new value of xp
+     */
+    public StatisticResult addExperience(Transaction transaction, int amount) {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord statisticsRecord = getStatistics(transaction);
+
+        int existingValue = statisticsRecord.getExperience();
+        int newValue = existingValue + amount;
+
+        statisticsRecord.setExperience(newValue);
+        statisticsRecord.store(KITPVP_STATISTICS.EXPERIENCE);
+
+        this.updateExperience(newValue);
+
+        return new StatisticResult(newValue);
+    }
+
+    /**
+     * Adds current killstreaks to the user account. Infallible.
+     *
+     * This will increment both current and highest killstreaks.
+     *
+     * @param transaction represents the current killstreak
+     * @param amount represents the amount of killstreak to add
+     * @return a result with the new value of killstreak
+     */
+    public StatisticResult addKillstreaks(Transaction transaction, int amount) {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord statisticsRecord = getStatistics(transaction);
+
+        int existingValue = statisticsRecord.getCurrentKillstreak();
+        int newValue = existingValue + amount;
+        int existingTwo = statisticsRecord.getHighestKillstreak();
+        int newTwo = existingTwo + amount;
+
+        statisticsRecord.setCurrentKillstreak(newValue);
+        statisticsRecord.setHighestKillstreak(newTwo);
```

Comment:
Why does this method unconditionally update the highest killstreak? What it should really do is update the highest killstreak if and only if the current killstreak is greater than the previous highest killstreak

Reply:
Will fix